[PT Run] [Uri plugin] Exception on query `^:`

### Microsoft PowerToys version

0.53.1

### Running as admin

- [ ] Yes

### Area(s) with issue?

PowerToys Run

### Steps to reproduce

Type the query: `^:`

### ✔️ Expected Behavior

No exception in Log.

### ❌ Actual Behavior

[PowerToysReport_2022-01-07-09-10-11.zip](https://github.com/microsoft/PowerToys/files/7827442/PowerToysReport_2022-01-07-09-10-11.zip)


```
-------------------------- Begin exception --------------------------
Message: Exception for plugin <Windows Uri Handler> when query <^:>

Exception full name  : System.UriFormatException
Exception message    : Invalid URI: The format of the URI could not be determined.
Exception stack trace:
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Uri..ctor(String uriString)
   at Microsoft.Plugin.Uri.UriHelper.ExtendedUriParser.TryParse(String input, Uri& result, Boolean& isWebUri)
   at Microsoft.Plugin.Uri.Main.Query(Query query)
   at PowerLauncher.Plugin.PluginManager.<>c__DisplayClass23_0.<QueryForPlugin>b__0()
   at Wox.Infrastructure.Stopwatch.Debug(String message, Action action)
   at PowerLauncher.Plugin.PluginManager.QueryForPlugin(PluginPair pair, Query query, Boolean delayedExecution)
Exception source     : System.Private.Uri
Exception target site: Void CreateThis(System.String, Boolean, System.UriKind)
Exception HResult    : -2146233033
-------------------------- End exception --------------------------
```

### Other Software

_No response_

![image](https://user-images.githubusercontent.com/61519853/148526535-c37ef5f1-0096-4ddb-b4bd-d8bd0479aaf1.png)

```
[2022-01-07 10:51:59.8883] [ERROR] [C:\a\_work\1\s\src\modules\launcher\PowerLauncher\Plugin\PluginManager.cs::220]
-------------------------- Begin exception --------------------------
Message: Exception for plugin <Windows Uri Handler> when query <^:>

Exception full name  : System.UriFormatException
Exception message    : Invalid URI: The format of the URI could not be determined.
Exception stack trace:
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Uri..ctor(String uriString)
   at Microsoft.Plugin.Uri.UriHelper.ExtendedUriParser.TryParse(String input, Uri& result, Boolean& isWebUri)
   at Microsoft.Plugin.Uri.Main.Query(Query query)
   at PowerLauncher.Plugin.PluginManager.<>c__DisplayClass23_0.<QueryForPlugin>b__0()
   at Wox.Infrastructure.Stopwatch.Debug(String message, Action action)
   at PowerLauncher.Plugin.PluginManager.QueryForPlugin(PluginPair pair, Query query, Boolean delayedExecution)
Exception source     : System.Private.Uri
Exception target site: Void CreateThis(System.String, Boolean, System.UriKind)
Exception HResult    : -2146233033
-------------------------- End exception --------------------------
```

---

**There is a second exception from indexer every time I type the query `^:`**
Not sure if it has the same cause.

```
[2022-01-07 10:53:27.7205] [ERROR] [C:\a\_work\1\s\src\modules\launcher\Plugins\Microsoft.Plugin.Indexer\Main.cs::167]
-------------------------- Begin exception --------------------------
Message: Something failed

Exception full name  : System.Data.OleDb.OleDbException
Exception message    : IErrorInfo.GetDescription failed with E_FAIL(0x80004005).
Exception stack trace:
   at System.Data.OleDb.OleDbCommand.ExecuteCommandTextErrorHandling(OleDbHResult hr)
   at System.Data.OleDb.OleDbCommand.ExecuteCommandTextForSingleResult(tagDBPARAMS dbParams, Object& executeResult)
   at System.Data.OleDb.OleDbCommand.ExecuteCommandText(Object& executeResult)
   at System.Data.OleDb.OleDbCommand.ExecuteCommand(CommandBehavior behavior, Object& executeResult)
   at System.Data.OleDb.OleDbCommand.ExecuteReaderInternal(CommandBehavior behavior, String method)
   at System.Data.OleDb.OleDbCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Plugin.Indexer.SearchHelper.OleDBSearch.Query(String connectionString, String sqlQuery)
   at Microsoft.Plugin.Indexer.SearchHelper.WindowsSearchAPI.ExecuteQuery(ISearchQueryHelper queryHelper, String keyword)
   at Microsoft.Plugin.Indexer.SearchHelper.WindowsSearchAPI.Search(String keyword, ISearchManager manager, String pattern, Int32 maxCount)
   at Microsoft.Plugin.Indexer.Main.Query(Query query, Boolean isFullQuery)
Exception source     : System.Data.OleDb
Exception target site: Void ExecuteCommandTextErrorHandling(System.Data.OleDb.OleDbHResult)
Exception HResult    : -2147215867
-------------------------- End exception --------------------------
```

--- a/src/modules/launcher/Plugins/Microsoft.Plugin.Uri.UnitTests/UriHelper/ExtendedUriParserTests.cs
+++ b/src/modules/launcher/Plugins/Microsoft.Plugin.Uri.UnitTests/UriHelper/ExtendedUriParserTests.cs
@@ -78,6 +78,7 @@ namespace Microsoft.Plugin.Uri.UnitTests.UriHelper
         [DataRow("ftp://user:password@google.com:2121", true, "ftp://user:password@google.com:2121/", false)]
         [DataRow("ftp://user:password@1.1.1.1", true, "ftp://user:password@1.1.1.1/", false)]
         [DataRow("ftp://user:password@1.1.1.1:2121", true, "ftp://user:password@1.1.1.1:2121/", false)]
+        [DataRow("^:", false, null, false)]
 
         public void TryParseCanParseHostName(string query, bool expectedSuccess, string expectedResult, bool expectedIsWebUri)
         {

--- a/src/modules/launcher/Plugins/Microsoft.Plugin.Uri/UriHelper/ExtendedUriParser.cs
+++ b/src/modules/launcher/Plugins/Microsoft.Plugin.Uri/UriHelper/ExtendedUriParser.cs
@@ -4,6 +4,7 @@
 
 using System;
 using System.Linq;
+using System.Text.RegularExpressions;
 using Microsoft.Plugin.Uri.Interfaces;
 
 namespace Microsoft.Plugin.Uri.UriHelper
@@ -21,10 +22,13 @@ namespace Microsoft.Plugin.Uri.UriHelper
 
             // Handling URL with only scheme, typically mailto or application uri.
             // Do nothing, return the result without urlBuilder
+            // And check if scheme match REC3986 (issue #15035)
+            const string schemeRegex = @"^([a-z][a-z0-9+\-.]*):";
             if (input.EndsWith(":", StringComparison.OrdinalIgnoreCase)
                 && !input.StartsWith("http", StringComparison.OrdinalIgnoreCase)
                 && !input.Contains("/", StringComparison.OrdinalIgnoreCase)
-                && !input.All(char.IsDigit))
+                && !input.All(char.IsDigit)
+                && Regex.IsMatch(input, schemeRegex))
             {
                 result = new System.Uri(input);
                 isWebUri = false;

